specification/azurelargeinstance/resource-manager/Microsoft.AzureLargeInstance/preview/2023-07-20-preview/azurelargeinstance.json

```diff
@@ -0,0 +1,1312 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureLargeInstance/azureLargeInstance/{azureLargeInstanceName}": {
+      "get": {
+        "tags": [
+          "Azure Large Instances"
+        ],
+        "operationId": "AzureLargeInstance_Get",
+        "summary": "Gets an Azure Large Instance.",
+        "description": "Gets an Azure Large Instance for the specified subscription, resource group, and instance name.",
+        "x-ms-examples": {
+          "Get an Azure Large Instance": {
+            "$ref": "./examples/AzureLargeInstance_Get.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AzureLargeInstanceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AzureLargeInstance"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Azure Large Instances"
+        ],
+        "operationId": "AzureLargeInstance_Delete",
+        "summary": "Deletes an Azure Large Instance.",
+        "description": "Deletes an Azure Large Instance for the specified subscription, resource group, and instance name.",
+        "x-ms-examples": {
+          "Delete an Azure Large Instance": {
+            "$ref": "./examples/AzureLargeInstance_Delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AzureLargeInstanceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The Azure Large Instance is deleted."
+          },
+          "204": {
+            "description": "NoContent"
+          },
+          "501": {
```

Comment:
A couple issues here...
-You aren't really encouraged to document all the possible HTTP error codes. It should 'go without saying' that any HTTP API can return HTTP error codes.
-501 is 'Not Implemented'. It feels a little contradictory to document that you might return this, on an API, but that otherwise it might succeed. Is the operation implemented or not? Does it depend on the request somehow?

IF you need to document this scenario, what I'd suggest doing instead, is not documenting the 501 status code specifically, but documenting it (with its expected statusCode etc) as part of the description of the 'default' error response, this would be more similar to what many other RPs are in order to document details of all the possible error statuses they can return _beyond_ the http status code. (Remember that there's all these detailed error status fields in the response body itself.)

Reply:
The put and delete operations are not implemented. As a part of the requirement for GA, I need to return an explicit response for the methods. 
Thanks, I'll document it with the default error.

Reply:
Are they not implemented because this resource type will never actually be created through ARM? But they are only created in some other system? If so, it might be better NOT to document them.

Reply:
I decided not to document the put and delete operations. I'm getting these errors - https://github.com/Azure/azure-rest-api-specs/pull/25221/checks?check_run_id=16120080828, but it seems like this isn't a hard requirement for the PR to be merged?

Reply:
Swagger lint diff operation hasn't synced with the pipeline. Can someone reset it?

Reply:
Removing it causes it to fail lint diff here - https://microsoft-my.sharepoint.com/:w:/p/ajeyaseelan/ETYX7Lk8E5pMlLDd-0oq--0BkrVo4jHZNYsGsxIycTtyrQ?e=QIfQ08 because a PUT is required. Should I just add the default error message instead?